WordPress/Classes/Extensions/String+Helpers.swift

```diff
@@ -10,8 +10,24 @@ extension String {
         return NSString.encodeXMLCharactersIn(self)
     }
 
+    func trim() -> String {
+        return stringByTrimmingCharactersInSet(NSCharacterSet.whitespaceAndNewlineCharacterSet())
+    }
+
     /// Returns `self` if not empty, or `nil` otherwise
     func nonEmptyString() -> String? {
         return isEmpty ? nil : self
     }
+
+    /// Remove the specified suffix from the string and returns the new string. 
+    ///
+    /// - Parameters: 
+    ///     - suffix: The suffxing to remove.  This can be a regular expression but should not include an ending `$`.
+    ///
+    func trimSuffix(suffix: String) -> String {
```

Comment:
The documentation does say that it expects a regex, but it can be easy to miss. I'd use a named parameter so it's obvious `string.trimSuffix(regexp: "wp-admin/?")`. Otherwise I'd expect it to do an exact string match